soweego/linker/evaluate.py

```diff
@@ -9,55 +9,114 @@
 __license__ = 'GPL-3.0'
 __copyright__ = 'Copyleft 2019, Hjfocs'
 
+import json
 import logging
 import os
+from collections import defaultdict
 
 import click
 import recordlinkage as rl
 from numpy import mean, std
 from pandas import concat
-from sklearn.model_selection import StratifiedKFold, train_test_split
+from sklearn.externals import joblib
+from sklearn.model_selection import (GridSearchCV, StratifiedKFold,
+                                     train_test_split)
 
-from soweego.commons import constants, target_database
+from soweego.commons import constants, target_database, utils
 from soweego.linker import train, workflow
 
 LOGGER = logging.getLogger(__name__)
 
 
-@click.command()
+@click.command(context_settings={'ignore_unknown_options': True, 'allow_extra_args': True})
 @click.argument('classifier', type=click.Choice(constants.CLASSIFIERS))
 @click.argument('target', type=click.Choice(target_database.available_targets()))
 @click.argument('target_type', type=click.Choice(target_database.available_types()))
+@click.option('--nested', is_flag=True, help='Compute a nested cross-validation with hyperparameters tuning via grid search.')
 @click.option('--single', is_flag=True, help='Compute a single evaluation over all k folds, instead of k evaluations.')
 @click.option('-k', '--k-folds', default=5, help="Number of folds, default: 5.")
-@click.option('-b', '--binarize', default=0.1, help="Default: 0.1.")
+@click.option('-m', '--metric', type=click.Choice(constants.PERFORMANCE_METRICS),
+              default='f1',
+              help="Performance metric for nested cross-validation. Implies '--nested'. Default: f1.")
 @click.option('-d', '--dir-io', type=click.Path(file_okay=False), default=constants.SHARED_FOLDER, help="Input/output directory, default: '%s'." % constants.SHARED_FOLDER)
-def cli(classifier, target, target_type, single, k_folds, binarize, dir_io):
-    """Evaluate the performance of a probabilistic linker."""
-    if not single:
-        predictions, p_mean, p_std, r_mean, r_std, fscore_mean, fscore_std = average_k_fold(
-            constants.CLASSIFIERS[classifier], target, target_type, binarize, dir_io, k=k_folds)
-        LOGGER.info('Precision: mean = %s; std = %s; Recall: mean = %s; std = %s; F-score: mean = %s; std = %s',
-                    p_mean, p_std, r_mean, r_std, fscore_mean, fscore_std)
+@click.pass_context
+def cli(ctx, classifier, target, target_type, nested, single, k_folds, metric, dir_io):
+    """Evaluate the performance of a probabilistic linker.
+    By default, it runs 5-fold cross-validation and returns averaged performance scores.
+    """
+    kwargs = utils.handle_extra_cli_args(ctx.args)
+    if kwargs is None:
+        return 1
```

Comment:
`sys.exit` should be used to specify the exit code of an application.

```suggestion
        sys.exit(1)
```